### DISCUSSION TITLE
How to calculate transits?

### DISCUSSION THREAD

**Discussion by jonx18** [g-battaglia/kerykeion#76] [Q&A]
Hi, I love your development, could you share with me how to calculate a transits chart? Thanks :D

**Comment by tutucea** [comment]
```
from skyfield.api import load

# Create a timescale and ask the current time.
ts = load.timescale()
t = ts.now()

# Load the JPL ephemeris DE421 (covers 1900-2050).
planets = load('de421.bsp')
earth, mars = planets['earth'], planets['mars']

# What's the position of Mars, viewed from Earth?
astrometric = earth.at(t).observe(mars)
ra, dec, distance = astrometric.radec()

print(ra)
print(dec)
print(distance)
```

try that for tracking mars ... If you click on my name i made a post just about the transits ..

**Comment by g-battaglia** [comment]
Hi @jonx18 ,
here's an example for the transit of the current day:

```python
from kerykeion import AstrologicalSubject, KerykeionChartSVG

subject = AstrologicalSubject("Kanye", 1977, 6, 8, 8, 45, "Atlanta", "USA")

# Astrological Subject for the current day
transit_day = AstrologicalSubject()

transit_chart = KerykeionChartSVG(subject, "Transit", transit_day)

transit_chart.makeSVG()
```

Please note that if you initialize an AstrologicalSubject without parameters, it is set to the current day.

**Reply by jonx18** [reply]
Thank you so much for the detailed explanation

**Reply by obertetti** [reply]
Hi @g-battaglia, love the project too, thanks so much for your hard work on this!

Once you have generated today's transit, can you assess the aspects between transiting planets and natal planets (e.g Jupiter trine Sun)? 

I've tried using the NatalAspects function applied to today's transit but doesn't seem to be generating the right results. 

Please see code below.
```
from kerykeion import AstrologicalSubject, KerykeionChartSVG
subject = AstrologicalSubject("Kanye", 1977, 6, 8, 8, 45, "Atlanta", "USA")
transits_today = AstrologicalSubject(name = "Kanye", city="Atlanta", nation = "USA") 

aspects_today = NatalAspects(transits_today)
relevant_aspects= aspects_today.relevant_aspects
```
